Automattic/jetpack
# Admin Page: Run class codemod on the admin page codebase

Fixes more warnings about createclass usage.

#### Changes proposed in this Pull Request:

* Runs the `class` codemod over `_inc/client` making 63 components not depend on `React.createClass`
* Runs `eslint --fix` on the files

For simplicity, files unprocessed will be addressed by another PR.

#### Testing instructions:

1. Check this branch
2. Build the Admin Page (`yarn build`)
3. Expect it to build properly.
4. Visit the Admin Page. Expect to see no warnings about using createClass from the main `React` package. ` Accessing createClass via the main React package is deprecated, and will be removed in React v16.0. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class v15.* is available on npm as a temporary, drop-in replacement`

#### Skipped files
```
_inc/client/components/gridicon/index.jsx: `Gridicon` was skipped because of inconvertible mixins.
_inc/client/components/info-popover/index.jsx: `` was skipped because of deprecated API calls. Remove calls to getDOMNode, isMounted, replaceProps, replaceState, setProps in your React component and re-run this script.
_inc/client/components/chart/legend.jsx: `LegendItem` was skipped because of inconvertible mixins.
_inc/client/components/count/index.jsx: `` was skipped because of inconvertible mixins.
_inc/client/components/external-link/index.jsx: `` was skipped because of inconvertible mixins.
_inc/client/components/form/input-checkbox-multiple.jsx: `` was skipped because of inconvertible mixins.
_inc/client/components/form/input-checkbox.jsx: `` was skipped because of inconvertible mixins.
_inc/client/components/form/input-hidden.jsx: `` was skipped because of inconvertible mixins.
_inc/client/components/form/input-radio.jsx: `` was skipped because of inconvertible mixins.
_inc/client/components/form/input-select.jsx: `` was skipped because of inconvertible mixins.
_inc/client/components/form/input-text.js: `` was skipped because of inconvertible mixins.
_inc/client/components/section-nav/index.jsx: `SectionNav` was skipped because `arguments` was found in your functions. Arrow functions do not expose an `arguments` object; consider changing to use ES6 spread operator and re-run this script.
_inc/client/components/button-group/docs/example.jsx: `Buttons` was skipped because of inconvertible mixins.
_inc/client/components/count/docs/example.jsx: `` was skipped because of inconvertible mixins.
_inc/client/components/section-nav/item.jsx: `NavItem` was skipped because of inconvertible mixins.
_inc/client/components/info-popover/docs/example.jsx: `InfoPopoverExample` was skipped because of inconvertible mixins.
_inc/client/components/popover/docs/example.jsx: `Popovers` was skipped because of inconvertible mixins.
_inc/client/components/search/docs/example.jsx: `SearchDemo` was skipped because of inconvertible mixins.
_inc/client/components/select-dropdown/docs/example.jsx: `SelectDropdownDemo` was skipped because of inconvertible mixins.
_inc/client/components/section-nav/docs/example.jsx: `SectionNavigation` was skipped because of inconvertible mixins.
_inc/client/components/spinner/docs/example.jsx: `` was skipped because of inconvertible mixins.
```
